### ISSUE TITLE
Literai - A website for sharing NaNoGenMo Stories and More

### ISSUE THREAD

**Issue by myles-oneill** [NaNoGenMo/2016#120]
For NaNoGenMo this year I decided to work with some friends to create something for the community. We created a new website, Literai:

https://www.literai.com

Literai is a place to share artificial fiction with a broad audience. I hope you consider posting your final novels on it to share them with the world.

We also built out some detailed tutorials (https://www.literai.com/tutorial/) to use Char-RNN as a way to generate artificial fiction relatively painlessly. We hope you try it out!

I'll be using the tutorial on our site to finish my own NaNoGenMo story for this year :)

**Comment by ikarth** [comment]
More tutorials are always a welcome sight.

**Comment by enkiv2** [comment]
I think you have a problem with your input validation. I've uploaded a
story (Seven Bad Heists) twice, and each time the result is a completely
empty story & method section with my ability to edit disabled.

**Comment by myles-oneill** [comment]
@enkiv2 Thanks for uploading your stories! It definitely looks like something is buggy with the uploader, we're going to look into it and try and fix it. For now I've manually deleted the broken stories.

**Comment by enkiv2** [comment]
Thanks. Let me know when it's repaired :)

(The only thing that the story that wasn't uploading properly had that
others didn't is percent signs, as far as I can tell.)

**Comment by myles-oneill** [comment]
@enkiv2 Was the length of that one particularly long compared to the others? The ones you posted might have already been pushing our uploader a bit in terms of size, so that is one theory I have as to why it broke down. If you could email me the file it might help with testing (myles.oneill@gmail.com).

**Comment by enkiv2** [comment]
The file in question is
https://github.com/enkiv2/NaNoGenMo-2015/blob/master/badHeists.md -- it's
only 238k.

**Comment by myles-oneill** [comment]
@enkiv2  We looked into it and it turns out its because we have a bug in the way we are escaping the text which causes the inclusion of the % sign to break the page. We're going to figure out a fix, but in the meantime if you'd like to post your story you could replace all instances of it in the text with the word "percent", not ideal obviously haha.

**Comment by enkiv2** [comment]
Sounds good! I figured it was something like that :)

**Comment by revereche** [comment]
Hey, checking out your tutorial for Windows - you can use Docker with VirtualBox instead, which is free and quite compatible with Windows Home editions!

**Comment by myles-oneill** [comment]
@revereche Awesome, I didn't know about that! I'll add it to the tutorial for others :)

**Comment by serin-delaunay** [comment]
My [story](https://github.com/serin-delaunay/NaNoGenMo2016/blob/master/output/data_leak_v2.md) is in markdown format. When I copy the text from github's (or dillinger.io's) HTML render into Literai's uplaod form, the bold formatting is lost (but the code formatting is preserved). When I copy the raw text into the upload form, Literai doesn't apply markdown formatting. Is there a way around this? I don't intend to manually set all the bold text using the upload form.

**Comment by myles-oneill** [comment]
@serin-delaunay We are using an open source rich text editor and I'm not super familiar with its quirks. It looks to me like its something fiddly about the dillinger renderer. I used this one from raw markdown and copying from it's HTML Preview copied the formatting correctly: http://daringfireball.net/projects/markdown/dingus

**Comment by serin-delaunay** [comment]
Thanks, that fixed it. Unfortunately there are percent signs in some of the passwords, so it broke the same way as @enkiv2's.

**Comment by myles-oneill** [comment]
@serin-delaunay We really need to fix that issue. I manually removed the 2 % signs from your story so it will display properly in the meantime. Thanks for sharing it!

**Comment by serin-delaunay** [comment]
Thanks! I didn't think to save the Methods section before hitting submit, so I'm relieved I don't have to type it again :D The `%`s were cipher `x`s, so I've edited `x`s in for now.

I copied some rich text from the story into the Methods section, and it produced a change in text size that was visible in the viewing page, but not the editing page. I've managed to clean it up, but it might be a good idea to have text size visible and alterable on the edit page if it can vary on the viewing page.

**Comment by myles-oneill** [comment]
Good idea, we need to spend a bit of time fixing up the rich text editing to make it cleaner.
